oci/archive/oci_transport.go

```diff
@@ -0,0 +1,179 @@
+// reference.Transport().ParseReference(reference.StringWithinTransport()) returns an equivalent reference.
+func (ref ociArchiveReference) StringWithinTransport() string {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.StringWithinTransport()
+}
+
+// DockerReference returns a Docker reference associated with this reference
+func (ref ociArchiveReference) DockerReference() reference.Named {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.DockerReference()
+}
+
+// PolicyConfigurationIdentity returns a string representation of the reference, suitable for policy lookup.
+func (ref ociArchiveReference) PolicyConfigurationIdentity() string {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.PolicyConfigurationIdentity()
+}
+
+// PolicyConfigurationNamespaces returns a list of other policy configuration namespaces to search
+// for if explicit configuration for PolicyConfigurationIdentity() is not set
+func (ref ociArchiveReference) PolicyConfigurationNamespaces() []string {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.PolicyConfigurationNamespaces()
+}
+
+// NewImage returns a types.Image for this reference, possibly specialized for this ImageTransport.
+// The caller must call .Close() on the returned Image.
+func (ref ociArchiveReference) NewImage(ctx *types.SystemContext) (types.Image, error) {
+	src, err := newImageSource(ref, ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	return image.FromSource(src)
+}
+
+// NewImageSource returns a types.ImageSource for this reference,
+// asking the backend to use a manifest from requestedManifestMIMETypes if possible.
+// The caller must call .Close() on the returned ImageSource.
+func (ref ociArchiveReference) NewImageSource(ctx *types.SystemContext, requestedManifestMIMETypes []string) (types.ImageSource, error) {
+	return newImageSource(ref, ctx, requestedManifestMIMETypes)
+}
+
+// NewImageDestination returns a types.ImageDestination for this reference.
+// The caller must call .Close() on the returned ImageDestination.
+func (ref ociArchiveReference) NewImageDestination(ctx *types.SystemContext) (types.ImageDestination, error) {
+	return newImageDestination(ref, ctx)
+}
+
+// DeleteImage deletes the named image from the registry, if supported.
+func (ref ociArchiveReference) DeleteImage(ctx *types.SystemContext) error {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.DeleteImage(ctx)
```

Comment:
This would delete the just-created temporary directory, not the archive file.

Luckily `layout.*.DeleteImage` just fails; this should probably as well.  Really deleting a single reference means deleting that entry in the index, and perhaps the referenced manifest and blob, it should not delete the whole archive.